### PR TITLE
Update Gatsby and all other packages

### LINKED ISSUE #163: Update site from Gatsby 2 to 4

### PR DESCRIPTION
**Pull request by rogermparent**
This PR is primarily meant to upgrade cml.dev to Gatsby 4, but in the process upgrades all other packages to their latest versions and does any changes needed to make builds work after.

It was a surprisingly painless process (not having a blog helps), but it's possible some small things will end up broken so we'll have to check it over before merging.

Fixes #163

### CODE REVIEW

**Comment by yathomasi** [comment]
Great @rogermparent. I checked thoroughly, it's running on my machine without any issues, and also the Heroku deployment is also looking good. I think we can proceed further.

**Comment by rogermparent** [comment]
Just fixed images, I think this PR is good to go!

And unlike dvc.org, the build doesn't break with existing cache- or maybe it's just so short that a second build doesn't time out.

**Review by casperdcl — APPROVED** [review]
yep images lgtm now :)

**Comment by rogermparent** [comment]
Fixed a minor style issue, code blocks in the title were a bit bigger due to changes in the GitHub Markdown CSS package and the longer ones were line-breaking.

before:
before

after:
after

File: .prettierrc [hunk 1]
```diff
@@ -6,5 +6,13 @@
   "printWidth": 80,
   "tabWidth": 2,
   "useTabs": false,
-  "proseWrap": "always"
+  "proseWrap": "always",
+  "overrides": [
+    {
+      "files": "*.css",
+      "options": {
+        "singleQuote": false
```
Comment (julieg18) [inline .prettierrc:14]: We should look into adding this to dvc.org!

File: gatsby-config.js [hunk 2]
```diff
@@ -49,6 +49,13 @@ const plugins = [
       path: path.join(__dirname, 'content')
     }
   },
+  {
+    resolve: 'gatsby-source-filesystem',
+    options: {
+      name: 'images',
+      path: path.join(__dirname, 'static', 'img')
+    }
+  },
```
Comment (julieg18) [inline gatsby-config.js:58]: Glad we're using `gatsby-plugin-image` for our docs! The images look kind of blurry. If I remember correctly from us fixing this with dvc.org, we need to add `quality: 90` to our `gatsby-remark-images` options.

**Review by julieg18 — COMMENTED** [review]
Great job @rogermparent! It looks great! Just one small thing about the docs images and this should be good to merge!

File: .prettierrc (hunk 1 above)
Comment (rogermparent) [reply .prettierrc:14]: dvc.org took a different approach by changing `format-staged` not to target css files, but giving it this configuration will probably make linting more robust.

File: gatsby-config.js (hunk 2 above)
Comment (rogermparent) [reply gatsby-config.js:58]: I remember the same solution to this problem, but after implementing it I don't notice any improvement. :thinking:

**Comment by rogermparent** [comment]
I'm going to merge this in since we're 99% of the way to parity with the previous site, any remaining issues like thumbnail image quality can be addressed in further iterations.

**Pull request merged**